[docs] v1.13.0 changelog (#6063)
DevExpress/testcafe: docs/articles/blog/2021-3-03-testcafe-v1-13-0-released.md

---
layout: post
title: TestCafe v1.13.0 Released
permalink: /blog/:title.html
---
# TestCafe v1.13.0 Released

This release adds support for custom paths to the configuration file, support for Microsoft Edge on Linux systems, and multiple bugfixes.

<!--more-->

## Enhancements

### ⚙ Specify Custom Path to the TestCafe Configuration File ([PR #6035](https://github.com/DevExpress/testcafe/pull/6035) by [@Chris-Greaves](https://github.com/Chris-Greaves))

TestCafe now enables you to specify a custom path to the [configuration file](../documentation/reference/configuration-file.md).

To set the path, use one of the following options:

* the [*--config-file* CLI option](../documentation/reference/command-line-interface.md#--config-file-path)
* the [*configFile* option of the *createTestCafe* function](../documentation/reference/testcafe-api/global/createtestcafe.md#options)

### Add Support for Microsoft Edge on Linux ([PR testcafe-browser-tools/#210](https://github.com/DevExpress/testcafe-browser-tools/pull/210) by [@dcsaszar](https://github.com/dcsaszar))

> If you follow the [Microsoft Edge Insider Channels for Linux](https://www.microsoftedgeinsider.com/en-us/download?platform=linux-deb) and have Microsoft Edge installed on your Linux machine, you can now launch TestCafe tests in this browser.

Reviewer: Hm, looks like you should ***follow*** a channel to launch TestCafe tests...

Would it be clearer to say that you can launch tests in a browser version from the Insider channel (if that's what you mean)?

Author: > Would it be clearer to say that you can launch tests in a browser version from the Insider channel?

Yes, that's right.

Is that better:

```suggestion
If you have the [Insider version of the Microsoft Edge](https://www.microsoftedgeinsider.com/en-us/download?platform=linux-deb) installed on your Linux machine, you can now launch TestCafe tests in this browser.
```

* TestCafe now closes if the `No tests to run` error occurs in Live mode ([#4257](https://github.com/DevExpress/testcafe/issues/4257))
* Fixed a freeze that happened when you run a test suite with skipped tests ([#4967](https://github.com/DevExpress/testcafe/issues/4967))
* Fixed an error where a `documentElement.transform.translate` call moved the TestCafe UI in the browser window ([#5606](https://github.com/DevExpress/testcafe/issues/5606))
* TestCafe now emits a warning if you pass an unawaited selector to an assertion ([#5554](https://github.com/DevExpress/testcafe/issues/5554))
* Fixed a crash that sometimes occurred in Chrome v85 and earlier on pages with scripts ([PR testcafe-hammerhead/#2590](https://github.com/DevExpress/testcafe-hammerhead/pull/2590))
* Overrode the [insertAdjacentElement method](https://developer.mozilla.org/en-US/docs/Web/API/Element/insertAdjacentElement) inside TestCafe ([testcafe-hammerhead/#1321](https://github.com/DevExpress/testcafe-hammerhead/issues/1321))
* Overrode the internal `Element.prototype.after` method ([testcafe-hammerhead/#1833](https://github.com/DevExpress/testcafe-hammerhead/issues/1833))

> * Overrode the [ParentNode.prepend method](https://developer.mozilla.org/en-US/docs/Web/API/ParentNode/prepend) inside TestCafe ([testcafe-hammerhead/#1005](https://github.com/DevExpress/testcafe-hammerhead/issues/1005))

Reviewer: Are these three fixes relevant to users?

Author: I think we can safely omit those
